algorithms_task_2/student/include/impl.h

```diff
@@ -9,4 +10,13 @@
  *
  * @return true if source starts from a prefix
  */
-bool startsWith(const std::string& source, const std::string& prefix);
+bool startsWith(const std::string& source, const std::string& prefix) {
+    if (prefix.empty()) {
+        return prefix.empty() || source.empty();
+    }
+    if (prefix.size() > source.length()) {
+        return false;
+    }
+
+    return std::equal(source.cbegin(), source.cbegin() + prefix.length(), prefix.cbegin());
```

Comment:
not the case here, but please make it safer, without any iterator shifts

Reply:
done